checkov/common/checks_infra/solvers/attribute_solvers/base_attribute_solver.py

```diff
@@ -199,11 +199,27 @@ def _is_variable_dependant(value: Any, source: str) -> bool:
         return False
 
     @staticmethod
-    def _render_json_str(value_to_check: Any, attr: str, vertex: Dict[str, Any]) -> Any:
-        if attr == 'policy' and vertex.get('resource_type', '').endswith('policy'):
-            try:
-                value_to_check = json.loads(value_to_check)
-                return value_to_check
-            except Exception as e:
-                logging.info(f'cant parse policy str to object, {str(e)}')
+    def _render_json_str(value_to_check: Any) -> Any:
+        """
+        Tries to render objects containing json dict, for example:
+        - resource "aws_iam_policy" "example1" {
+            policy == <<POLICY
+                {
+                ... json-data
+                }
+            POLICY
+        }
+        - resource "aws_emr_security_configuration" "example2" {
+            configuration = <<EOF
+                {
+                ... json-data
+                }
+            EOF
+        }
+        """
+        try:
+            value_to_check = json.loads(value_to_check)
```

Comment:
I don't think this is a good idea, because it will also change none JSON strings, like `"true" -> True` or `"12345" -> 12345`. We should at least do a string lookup, before trying to cast to dict/list.
Also not sure, if someone does it, but if someone actually wants the value to be a string, then we will now break it.
ex. `name = "[\"some name\"]"` so this wouldn't be a feature, it would be a breaking change.

Reply:
got you, my intention was we don't want to add support for each attribute terraform allow you to add as an inlne configuration (to reduce future maintenance cost).
Do you have any idea how to do it better?
If not, I will just regress it to the way it was before and add support for `configuration` as well